src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -239,4 +232,248 @@ else if (computer_battleship.scan(coor)){
     public boolean getScanResult() {
         return scanResult;
     }
+
+
+    
+    // hardMode: boolean difficulty state variable, true if the game is played in hard mode
+    private boolean hardMode = false;
+
+    
+    // setDifficulty: if user pressed the 'hard' button, then set state variable hardMode to true and call
+    // placeComputerShipsHard to place the computer's ships randomly
+    public void setDifficulty(String difficulty) {
+        if (difficulty.equals("hard")) {
+            hardMode = true;
+            placeComputerShipsHard();
+        }
+    }
+
+    
+    
+    // placeComputerShipsHard: if hard mode is selected, then the computer places ships in random locations across
+    // the board (with no overlapping)
+    private void placeComputerShipsHard() {
```

Comment:
Function is quite lengthy. Could use some clean-up. Consider breaking into multiple functions to reduce duplicate code.